controllers/chaosimpl/physicalmachinechaos/impl.go

```diff
@@ -0,0 +1,185 @@
+// Copyright 2021 Chaos Mesh Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package physicalmachinechaos
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	"github.com/go-logr/logr"
+	"go.uber.org/fx"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/chaos-mesh/chaos-mesh/api/v1alpha1"
+	"github.com/chaos-mesh/chaos-mesh/controllers/common"
+)
+
+type Impl struct {
+	client.Client
+	Log logr.Logger
+}
+
+func (impl *Impl) Apply(ctx context.Context, index int, records []*v1alpha1.Record, obj v1alpha1.InnerObject) (v1alpha1.Phase, error) {
+	impl.Log.Info("apply physical machine chaos")
+
+	physicalMachinechaos := obj.(*v1alpha1.PhysicalMachineChaos)
+	addresses := records[index].Id
+	addressArray := strings.Split(addresses, ",")
+
+	// for example, physicalMachinechaos.Spec.Action is 'network-delay', action is 'network', subAction is 'delay'
+	actions := strings.SplitN(string(physicalMachinechaos.Spec.Action), "-", 2)
+	if len(actions) != 2 {
+		err := errors.New("action invalid")
```

Comment:
I am not sure ,but "process" may be an invalid action here.

Reply:
Yeah, here is a bug, I fix it in https://github.com/chaos-mesh/chaos-mesh/pull/2235/commits/1586aabe3a0034a5a8c295bf15bd02983dbefce6